src/StateService/Verification/VerificationContext.cs

```diff
@@ -0,0 +1,136 @@
+using Akka.Actor;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.Plugins.StateService.Storage;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
+using System;
+using System.Collections.Generic;
+
+namespace Neo.Plugins.StateService.Verification
+{
+    public class VerificationContext
+    {
+        private const uint MaxValidUntilBlockIncrement = 100;
+        private StateRoot root;
+        private ExtensiblePayload payload;
+        private readonly Wallet wallet;
+        private readonly KeyPair keyPair;
+        private readonly int myIndex;
+        private readonly uint rootIndex;
+        private readonly ECPoint[] verifiers;
+        private int M => verifiers.Length - (verifiers.Length - 1) / 3;
+        private readonly Dictionary<int, byte[]> signatures = new Dictionary<int, byte[]>();
+
+        public bool IsValidator => myIndex >= 0;
+        public int MyIndex => myIndex;
+        public ECPoint[] Verifiers => verifiers;
+        public ICancelable Timer;
+        public StateRoot StateRoot
+        {
+            get
+            {
+                if (root is null)
+                {
+                    using var snapshot = StateStore.Singleton.GetSnapshot();
+                    root = snapshot.GetStateRoot(rootIndex);
+                }
+                return root;
+            }
+        }
+        public ExtensiblePayload Message => payload;
+
+        public VerificationContext(Wallet wallet, uint index)
+        {
+            this.wallet = wallet;
+            myIndex = -1;
+            root = null;
+            rootIndex = index;
+            using SnapshotCache snapshot = Blockchain.Singleton.GetSnapshot();
+            verifiers = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.StateValidator, index);
```

Comment:
I got it now. It will be automatic with those that are published with the `Role.StateValidator`.
This is exactly what I was expecting.

Now, I need to understand how to update it.

Reply:
Who are the first `StateValidator`, do the committee need to call `designateAsRole` or there will be default nodes such as `Validators`, `Committee` and `Oracles` as default MPT?

Reply:
I think committee need call `designateAsRole ` to designate validators.

Reply:
In my initial tests here I got an error after the Genesis.

Perhaps,as default we should have some node to do that,even if it is a single node.

Reply:
> In my initial tests here I got an error after the Genesis.

What error? Is it in this pr?

> Perhaps,as default we should have some node to do that,even if it is a single node.
> I mean,before the committee vote.

We should discuss this in neo-core.